### PR TITLE
feat: ability for disabling calls controls

### PR DESCRIPTION
**Pull request by shabanov-dlg**

### CODE REVIEW

File: src/components/CallControls/CallControls.js [hunk 1]
```diff
@@ -26,6 +27,13 @@ export type Props = {
 };
 
 class CallControls extends PureComponent<Props> {
+  isDisabledButton(id: string): boolean {
+    const { disabled } = this.props;
+    const isDisabled = typeof disabled === 'object' ? disabled[id] : disabled;
```
Comment (meafmira) [inline src/components/CallControls/CallControls.js]: Oh no no. This is way to check for `Object` type. Do you know, for example `typeof null === 'object'`. Also this is not right type for such goals. Let's replace `disabled?: boolean | { [string]: boolean }` with
```
disabled: boolean,
disabledButtonIds: Array<string>,
```
with default values `false` and `[]`

Comment (shabanov-dlg) [reply src/components/CallControls/CallControls.js]: I agree with checking for null. Array more informative structure for this.

But can you explain the logic you propose?
I have 2 cases:
1) disable all buttons
2) disable 1 button.

How can I disable 1 button this with 2 fields?

``
disabled: true, 
disabledButtonIds: ['button_id']
``
or
``
disabled: false, 
disabledButtonIds: ['button_id']
``
?

I think that 2 fields is redundant.

Comment (meafmira) [reply src/components/CallControls/CallControls.js]: `<button disabled={disabled && disabledIds.includes(buttonId)} />`

File: src/components/IconButton/IconButton.js [hunk 2]
```diff
@@ -19,7 +19,7 @@ export type Props = {
   flat: boolean,
   disabled: boolean,
   active?: boolean,
-  onClick: (event: SyntheticEvent<HTMLButtonElement>) => mixed,
+  onClick?: ?(event: SyntheticEvent<HTMLButtonElement>) => mixed,
```
[the remark this answers is not in the document]
Comment (shabanov-dlg) [reply src/components/IconButton/IconButton.js]: Flow checking fails

File: src/components/CallControls/CallControls.js (hunk 1 above)
Comment (shabanov-dlg) [reply src/components/CallControls/CallControls.js]: Logic is unclear.

 User will need to look into the code to figure out how to disable the button.
Please, check new version of my code.

File: src/components/IconButton/IconButton.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (meafmira) [reply src/components/IconButton/IconButton.js]: Where?

File: src/components/CallControls/CallControls.js (hunk 1 above)
Comment (meafmira) [reply src/components/CallControls/CallControls.js]: In your case user need to look into code too for see than `disabled` can be boolean or object

File: src/components/CallControls/CallControls.js [hunk 3]
```diff
@@ -14,6 +14,7 @@ export type Props = {
   size: 'normal' | 'large',
   isVisible: boolean,
   isMuted?: boolean,
+  disabled?: boolean | string[],
```
Comment (meafmira) [inline src/components/CallControls/CallControls.js]: Let's do this:
```
{
  ...,
  disabled: boolean,
} | {
  ...,
  disabledButtonIds: Array<string>,
}
```

File: src/components/IconButton/IconButton.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (shabanov-dlg) [reply src/components/IconButton/IconButton.js]: @meafmira https://circleci.com/gh/dialogs/dialog-web-components/2961

File: src/components/CallControls/CallControls.js (hunk 1 above)
Comment (shabanov-dlg) [reply src/components/CallControls/CallControls.js]: >In your case user need to look into code too for see than `disabled` can be boolean or object
It is semantic - it's normal:)
 
If I need to block all buttons, do I need to pass all id values?

File: src/components/CallControls/CallControls.js (hunk 3 above)
Comment (meafmira) [reply src/components/CallControls/CallControls.js]: @shabanov-dlg agree. This is more readable and exact semantic

Comment (shabanov-dlg) [reply src/components/CallControls/CallControls.js]: If you don't like the dynamic type - maybe I change name to "disabledAll" for boolean field?
And create all of this 2 fields not required?

**Pull request merged**